packages/data/src/components/use-select/index.js

```diff
@@ -78,7 +78,7 @@ export default function useSelect( _mapSelect, deps ) {
 	const mapSelect = useCallback( _mapSelect, deps );
 	const registry = useRegistry();
 	const isAsync = useAsyncMode();
-	const queueContext = useMemo( () => ( { queue: true } ), [ registry ] );
+	const queueContext = useMemo( () => ( { queue: true } ), [ registry, isAsync ] );
```

Comment:
This will make us lose previous callbacks when async mode changes.

Also see https://github.com/WordPress/gutenberg/issues/19199#issuecomment-568090443.